new_dawn_server/users/api/resources.py

```diff
@@ -255,6 +272,10 @@ def obj_create(self, bundle, **kwargs):
                 **self._get_model_fields_dict(bundle, ACCOUNT_FIELDS)
             )
             account.save()
+
+            # Deserialize City Preference Information and add to account
+            self.deserialize_city_pref(bundle, account)
```

Comment:
Try using this function to get city_pref, and directly pass that as a list like Account(city_preference=get_and_save_city_pref) 

(not sure if it works but should be able to work)

Reply:
It seems like for many-to-many fields, you need to use add() or set(). Otherwise, it will throw

TypeError: Direct assignment to the forward side of a many-to-many set is prohibited. Use city_preference.set() instead.

Reply:
I see. Then let's try city_preference right above account.save()